### PR TITLE
Allows you to offer an item to only one person with Shift+Ctrl+Click

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

You can click someone directly with shift+ctrl+click to offer an item only to them. This is in contrast with pressing G, which offers the item to every adjacent carbon mob.

Also fixes a runtime where the Give screen alert on a potential recipient was trying to remove itself on proximity loss to the giver after the Offering status effect had already done it.

Removes duplicate range check on Give screen alert that was causing the runtime as Offering status effect takes care of it.

Also adds a check after clicking the screen alert to take something to make sure we're not dead or incapacitated, so dead people can no longer take things.

Also adds a screentip for this functionality.

Also adds some more checks to give() to make sure we can do it before sending the message to players that we're offering something.

## Why It's Good For The Game

It always felt like you should be able to offer things to people directly, instead of anybody adjacent being able to take it. It felt like it wasn't done that way to avoid having to pick a name from a pop-up window to offer to, which is clunky and steals focus.

This makes it so people don't accidentally take something that wasn't intended for them and piss off the players involved. 

Also lowers incidences of theft, because you should be able to choose who you're giving something to without having to move to a tile with nobody else adjacent.

## Changelog

:cl:
expansion: Can Shift+Ctrl+Click people to give items to them directly now
fix: Fixed runtime on Give screen alert where a proximity check was happening twice
fix: Fixed being able to take things as a dead person
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/inventory.dm [hunk 1]
```diff
@@ -179,10 +179,10 @@
 	if(offered_item.on_offered(src)) // see if the item interrupts with its own behavior
 		return
 
-	visible_message(span_notice("[src] is offering [offered_item]."), \
-					span_notice("You offer [offered_item]."), null, 2)
+	visible_message(span_notice("[src] is offering [offered ? "[offered] " : ""][offered_item]."), \
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/inventory.dm:182]: This is gonna have double space if there’s no target

**Comment by bruhlookatthisdood** [comment]
now i can give an assistant a gun without the clown stealing it!

**Comment by SuperNovaa41** [comment]
> now i can give an assistant a gun without the clown stealing it!

You already could, just press G (its already bound by default)

**Comment by MrDoomBringer** [comment]
> You already could, just press G (its already bound by default)

The point of this PR is that G offers it to everyone nearby, wheras shift+ctrl+click is targetted

**Comment by bruhlookatthisdood** [comment]
> > now i can give an assistant a gun without the clown stealing it!
> 
> You already could, just press G (its already bound by default)

not when the clown is standing next to me ):<

**Comment by Colovorat** [comment]
> > You already could, just press G (its already bound by default)
> 
> The point of this PR is that G offers it to everyone nearby, wheras shift+ctrl+click is targetted

wait really??????????

**Comment by SuperNovaa41** [comment]
> > You already could, just press G (its already bound by default)
> 
> The point of this PR is that G offers it to everyone nearby, wheras shift+ctrl+click is targetted

Oh i see yes i missed that

**Comment by Colovorat** [comment]
Maybe change default offering to this one? So we only have one

**Comment by MrDoomBringer** [comment]
tbpqhph handing an item to someone else should have a risk of it being nabbed by a passer-by unless you make an effort to distance yourself from other people

**Comment by cacogen** [comment]
>tbpqhph handing an item to someone else should have a risk of it being nabbed by a passer-by unless you make an effort to distance yourself from other people

No, it shouldn't. You can still get the object from them using regular means.

**Comment by MrDoomBringer** [comment]
> No, it shouldn't. 

why not

**Comment by SkeletalElite** [comment]
Since that PR was merged I've literally never used this and only had someone use it to give me something once so I agree with this change

**Comment by cacogen** [comment]
>Yes it should

I think SkeletalElite raises a good point that you're never going to use it while adjacent to other people because of the possibility of theft, which leads to it being an annoying feature to use because you have to move away from other people every time.

**Comment by bluetshirtguy** [comment]
Shift+Ctrl+Click eh? all these new controls are begging for hand cramps man...

**Comment by SuperNovaa41** [comment]
> Since that PR was merged I've literally never used this and only had someone use it to give me something once so I agree with this change

I use it a LOT and I know many others that frequently use it too

**Comment by Farquaar** [comment]
1. I use the offer feature very often. I see it used multiple times a round, most of the time.
2. I think that it's probably a good thing that items can be snatched when they change hands. More opportunities for chaos and conflict.

**Comment by Dorsisdwarf** [comment]
People (G)ive me items all the time and i'm never able to reciprocate and have to dump it at my feet because I cant make the keybind work for unknown reasons so I support this

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -85,6 +85,12 @@
 
 	return ..()
 
+/mob/living/carbon/CtrlShiftClick(mob/user)
+	..()
+	if(iscarbon(user))
+		var/mob/living/carbon/carbon_user = user
+		carbon_user.give(src)
```
Comment (Watermelon914) [inline code/modules/mob/living/carbon/carbon.dm:92]: Put this on right click instead, have it only work if they're holding an item and they're not in combat mode

Comment (cacogen) [reply code/modules/mob/living/carbon/carbon.dm:92]: Doesn't this draw with a syringe? There are probably other right-click non-combat mode interactions I can think of. I don't think Mothblocks would approve of blocking off that interaction for all items.

Comment (Watermelon914) [reply code/modules/mob/living/carbon/carbon.dm:92]: ctrl + shift + click is a bit of an annoying combo, are you sure you can't just do shift + right click?

Comment (SkeletalElite) [reply code/modules/mob/living/carbon/carbon.dm:92]: shift right click opens the context menu

Comment (Watermelon914) [reply code/modules/mob/living/carbon/carbon.dm:92]: > shift right click opens the context menu

ah yeah, that's right, alright, whatever I suppose this is fine until an alternative can be found

**Pull request merged**